hadoop-tools/hadoop-azure/src/test/java/org/apache/hadoop/fs/azurebfs/ITestAbfsOutputStream.java

```diff
@@ -0,0 +1,278 @@
+    FileSystem.Statistics statistics = fs.getFsStatistics();
+    abfss.getAbfsConfiguration().setDisableOutputStreamFlush(false);
+    String testBytesToUpload = "bytes";
+
+    AbfsOutputStream outForSomeBytes = null;
+    try {
+      outForSomeBytes = (AbfsOutputStream) abfss.createFile(uploadBytesFilePath,
+          statistics,
+          true,
+          FsPermission.getDefault(), FsPermission.getUMask(fs.getConf()));
+
+      //Test for zero bytes To upload
+      assertValues("bytes to upload", 0,
+          outForSomeBytes.getOutputStreamStatistics().bytesToUpload);
+
+      outForSomeBytes.write(testBytesToUpload.getBytes());
+      outForSomeBytes.flush();
+
+      //Test for some bytes to upload
+      assertValues("bytes to upload", testBytesToUpload.getBytes().length,
+          outForSomeBytes.getOutputStreamStatistics().bytesToUpload);
+
+      //Test for relation between bytesUploadSuccessful, bytesUploadFailed
+      // and bytesToUpload
+      assertValues("bytesUploadSuccessful equal to difference between "
+              + "bytesToUpload and bytesUploadFailed",
+          outForSomeBytes.getOutputStreamStatistics().bytesUploadSuccessful,
+          outForSomeBytes.getOutputStreamStatistics().bytesToUpload -
+              outForSomeBytes.getOutputStreamStatistics().bytesUploadFailed);
+
+    } finally {
+      if (outForSomeBytes != null) {
+        outForSomeBytes.close();
+      }
+    }
+
+    AbfsOutputStream outForLargeBytes = null;
+    try {
+      outForLargeBytes =
+          (AbfsOutputStream) abfss.createFile(uploadBytesFilePath,
+              statistics
+              , true, FsPermission.getDefault(),
+              FsPermission.getUMask(fs.getConf()));
+
+      int largeValue = 100000;
+      for (int i = 0; i < largeValue; i++) {
+        outForLargeBytes.write(testBytesToUpload.getBytes());
+      }
+      outForLargeBytes.flush();
+
+      //Test for large bytes to upload
+      assertValues("bytes to upload",
+          largeValue * (testBytesToUpload.getBytes().length),
+          outForLargeBytes.getOutputStreamStatistics().bytesToUpload);
+
+      //Test for relation between bytesUploadSuccessful, bytesUploadFailed
+      // and bytesToUpload
+      assertValues("bytesUploadSuccessful equal to difference between "
+              + "bytesToUpload and bytesUploadFailed",
+          outForSomeBytes.getOutputStreamStatistics().bytesUploadSuccessful,
+          outForSomeBytes.getOutputStreamStatistics().bytesToUpload -
+              outForSomeBytes.getOutputStreamStatistics().bytesUploadFailed);
+
+    } finally {
+      if (outForLargeBytes != null) {
+        outForLargeBytes.close();
+      }
+    }
+
+  }
+
+  /**
+   * Tests to check time spend on waiting for tasks to be complete on a
+   * blocking queue in {@link AbfsOutputStream}.
+   *
+   * @throws IOException
+   */
+  @Test
+  public void testAbfsOutputStreamTimeSpendOnWaitTask() throws IOException {
```

Comment:
Need help on how to write tests for this counter.

Reply:
I don't see any easy way except to assert that it is > 0

Reply:
Also, "time spent"